src/Scrutor/AttributeSelector.cs

```diff
@@ -20,15 +21,41 @@ public void Populate(IServiceCollection services)
             {
                 var typeInfo = type.GetTypeInfo();
 
-                foreach (var attribute in typeInfo.GetCustomAttributes<ServiceDescriptorAttribute>())
+                var attributes = typeInfo.GetCustomAttributes<ServiceDescriptorAttribute>().ToArray();
+                // check whether they have the same service type
+                var duplicates = attributes.GroupBy(s => s).SelectMany(grp => grp.Skip(1));
+                if (duplicates.Any())
                 {
-                    var serviceType = attribute.ServiceType ?? type;
+                    throw new InvalidOperationException($"Type \"{type.FullName}\" has multiple ServiceDescriptors specified with the same service type.");
```

Comment:
I added this additional check. But I'm not sure whether it makes sense. Some of the unit tests related to `AsMatchingInterface()` fails after I add test data, so maybe this is already handled somehow?